pkg/db/migration/convert/convert000029/convert.go

```diff
@@ -45,7 +49,7 @@ func (c *ConvertToArtifactType) migratePipeline() error {
 			}
 
 			if isRecipeUpdated {
-				recipeYAML, err := yaml.Marshal(p.Recipe)
+				recipeYAML, err := yaml.Marshal(p.RecipeYAML)
```

Comment:
Hi @chuang8511,

1. Will `yaml.Marshal(p.RecipeYAML)` preserve all the comments and layout from the original YAML?
2. I have doubts about this working as expected since you're migrating `p.Recipe`, but you're saving `p.RecipeYAML`.

Reply:
Sorry, I did not notice this part. I will check it later.